src/Engine/Physics/Trigger.cpp

```diff
@@ -1,26 +1,30 @@
 #include "Trigger.hpp"
 
 #include <btBulletDynamicsCommon.h>
+#include "../Component/Physics.hpp"
 #include "RigidBody.hpp"
 #include "Shape.hpp"
 
 namespace Physics {
 
-    Trigger::Trigger(Shape* shape) : btCollisionWorld::ContactResultCallback() {
+    Trigger::Trigger(Component::Physics* comp) : btCollisionWorld::ContactResultCallback() {
         trigger = new btCollisionObject();
-        trigger->setCollisionShape(shape->GetShape());
+        
+        trigger->setCollisionShape(comp->GetShape().GetShape());
+        trigger->setWorldTransform(comp->GetRigidBody().GetRigidBody()->getWorldTransform());
     }
 
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'addSingleResult' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)